tests/test-autoupdater.c

```diff
@@ -237,6 +325,12 @@ main (int    argc,
   g_test_init (&argc, &argv, G_TEST_OPTION_ISOLATE_DIRS, NULL);
 
   eos_test_add ("/autoupdater/poll-results", NULL, test_poll_results);
+  UserVisibleTestData uvu_delay_data = { 1, FALSE, FALSE };
+  eos_test_add ("/autoupdater/user-visible-update/delay", &uvu_delay_data, test_user_visible_update_delay);
+  UserVisibleTestData uvu_nodelay_data = { 0, FALSE, TRUE };
+  eos_test_add ("/autoupdater/user-visible-update/nodelay", &uvu_nodelay_data, test_user_visible_update_delay);
+  UserVisibleTestData uvu_force_data = { 1, TRUE, TRUE };
+  eos_test_add ("/autoupdater/user-visible-update/force", &uvu_force_data, test_user_visible_update_delay);
```

Comment:
Nitpick: It might be clearer to use compound literals, since we’re always targeting a compiler which can support them:
```suggestion
  eos_test_add ("/autoupdater/user-visible-update/delay", (&(UserVisibleTestData) {
                  .update_delay = 1,
                  .force_update = FALSE,
                  .expected_has_commit = FALSE,
                }), test_user_visible_update_delay);
  eos_test_add ("/autoupdater/user-visible-update/nodelay", (&(UserVisibleTestData) {
                  .update_delay = 0,
                  .force_update = FALSE,
                  .expected_has_commit = TRUE,
                }), test_user_visible_update_delay);
  eos_test_add ("/autoupdater/user-visible-update/force", (&(UserVisibleTestData) {
                  .update_delay = 1,
                  .force_update = TRUE,
                  .expected_has_commit = TRUE,
                }), test_user_visible_update_delay);
```

Reply:
Wow, I was not aware you could do such a thing in C.

Reply:
Unfortunately, this gets miscompiled. I believe because `g_test_add` is a macro whose body is a block, the lifetime of the compound literal ends after the call to `eos_test_add`. By the time `g_test_run` iterates the table, the addresses are no longer valid. What's happening for me is that the last compound literal gets used for all the cases since the compiler is reusing the same address.

I'll play around with it a bit more, but I think I'll likely have to keep what I have.

Reply:
If I change `eos_test_add` to be a function instead of macro, it works.

Reply:
Oh, this has revealed a bug in https://github.com/endlessm/eos-updater/commit/11a34ee1625129b02909069db2740189a2bd0145 since the first 2 cases weren't actually being run. I'll split this part out.

Reply:
#322